docs/madr/decisions/046-meshservice-multizone.md

```diff
@@ -0,0 +1,360 @@
+  * MeshMultiZoneService
+  * MeshService with meshServiceSelector selector applied on global CP
+
+## Decision Outcome
+
+Chosen options:
+* Access service from another cluster use case
+  * MeshService synced to other zones, because it's the only way
+  * Export field, because it's more explicit and more managable
+* Replicated services use case
+  * MeshMultiZoneService, with potentially diverging to "MeshService with meshServiceSelector selector applied on global CP" based on the future policy matching MADR.
+
+## Pros and Cons of the Options
+
+### Access service from another cluster use case
+
+#### MeshService synced to other zones
+
+MeshService is an object created on the zone that matches a proxies in this specific zone.
+This object is the synced to the global control plane.
+
+To solve the first use case we can then sync MeshService to other zones.
+Let's say we have the following MeshService in zone `east`
+```yaml
+kind: MeshService
+metadata:
+  name: redis
+  namespace: redis-system
+labels:
+  kuma.io/display-name: redis
+  k8s.kuma.io/namespace: redis-system
+  kuma.io/zone: east
+spec:
+  selector:
+    dataplaneTags:
+      app: redis
+      k8s.kuma.io/namespace: redis-system
+  ports: ...
+status: ...
+```
+
+This will be synced to global and look like this
+```yaml
+kind: MeshService
+metadata:
+  name: redis-xyz12345 # we add hash suffix of (mesh, zone, namespace)
+  namespace: kuma-system # not that it's synced to system namespace
+  labels:
+    kuma.io/display-name: redis # original name of the object on the Kubernetes
+    k8s.kuma.io/namespace: redis-system
+    kuma.io/zone: east
+    kuma.io/origin: zone
+spec:
+  selector:
+    dataplaneTags:
+      app: redis
+      k8s.kuma.io/namespace: redis-system
+      kuma.io/zone: east-1
+  ports: ...
+status: ...
+```
+
+This will be then synced to other zones in this form, which is the same syncing strategy that we already have with ZoneIngress.
+
+Because of the hashes and syncing only to system namespace, we avoid clashes of synced services and existing services in the cluster.
+
+**Hostname and VIP management**
+
+Advertising the service is only part of the solution. We also need to provide a way to consume this service.
+
+We cannot reuse:
+* existing hostnames like `redis.redis-system.svc.cluster.local`, because there might be already a redis in cluster `east.
+* ClusterIP, because each cluster network is isolated. We might clash with IPs.
+
+Therefore `status` object should not be synced from zone `east` to zone `west`.
+
+To provide hostname, we can use `HostnameGenerator` described in MADR #XXX (todo provide a number when merged).
+```yaml
+type: HostnameGenerator
+name: k8s-zone-hostnames
+spec:
+  template: {{ label "kuma.io/display-name" }}.{{ label "k8s.kuma.io/namespace" }}.svc.cluster.{{ label "kuma.io/zone" }}
+```
+so that the synced service in zone `west` gets this hostname `redis.redis-system.svc.cluster.east`.
```

Comment:
This imply we should suggest to stop using `local` as a zone.

Also we said `mesh` instead of `cluster` no?

Reply:
> This imply we should suggest to stop using local as a zone.

I don't think so. I think you should Kube hostnames (`.local`) if you can. In case of emergency if you want to roll out mesh from a couple of services, you can do this. If you change hostnames, you can't.

> Also we said mesh instead of cluster no?

Yes, let me fix it